src/main/java/seedu/clinic/model/attribute/NameContainsKeywordsPredicateForSupplier.java

```diff
@@ -18,8 +18,16 @@ public NameContainsKeywordsPredicateForSupplier(List<String> keywords) {
 
     @Override
     public boolean test(Supplier supplier) {
-        return keywords.stream()
-                .anyMatch(keyword -> StringUtil.containsWordIgnoreCase(supplier.getName().fullName, keyword));
+        boolean testValue;
+        try {
+            testValue = keywords.stream()
+                    .anyMatch(keyword ->
+                            StringUtil.containsWordIgnoreCase(supplier.getName().fullName, keyword));
+        } catch (IllegalArgumentException ex) {
+            testValue = keywords.stream()
+                    .anyMatch(keyword -> supplier.getName().fullName.equals(keyword));
```

Comment:
Maybe it's better to handle the exception within the stringUtil.containsWordIgnoreCase?